pkg/nerdgraph/typegen.go

```diff
@@ -0,0 +1,223 @@
+package nerdgraph
+
+import (
+	"fmt"
+	"strings"
+)
+
+func ResolveSchemaTypes(schema Schema, typeNames []string) (map[string]string, error) {
+	typeKeeper := make(map[string]string)
+
+	for _, typeName := range typeNames {
+		typeGenResult, err := TypeGen(schema, typeName)
+		if err != nil {
+			fmt.Printf("ERROR while generating type %s: %s", typeName, err)
+		}
+
+		for k, v := range typeGenResult {
+			typeKeeper[k] = v
+		}
+	}
+
+	return typeKeeper, nil
+}
+
+func handleEnumType(schema Schema, t SchemaType) map[string]string {
+	types := make(map[string]string)
+
+	// output collects each line of a struct type
+	output := []string{}
+	output = append(output, fmt.Sprintf("type %s int ", t.Name))
+	output = append(output, "")
+
+	output = append(output, "const (")
+	for i, v := range t.EnumValues {
+		if i == 0 {
+			output = append(output, fmt.Sprintf("\t%s = iota", v.Name))
+		} else {
+			output = append(output, fmt.Sprintf("\t%s", v.Name))
+		}
+
+	}
+	output = append(output, ")")
+	output = append(output, "")
+
+	types[t.Name] = strings.Join(output, "\n")
+
+	return types
+}
+
+// firstTypeName returns the first non-empty name in the type tree that is found.
+func firstTypeName(f SchemaInputValue) string {
+	if f.Type.Name != "" {
+		return f.Type.Name
+	} else if f.Type.OfType.Name != "" {
+		return f.Type.OfType.Name
+	} else if f.Type.OfType.OfType.Name != "" {
+		return f.Type.OfType.OfType.Name
+	} else if f.Type.OfType.OfType.OfType.Name != "" {
+		return f.Type.OfType.OfType.OfType.Name
+	} else if f.Type.OfType.OfType.OfType.OfType.Name != "" {
+		return f.Type.OfType.OfType.OfType.OfType.Name
+	} else if f.Type.OfType.OfType.OfType.OfType.OfType.Name != "" {
+		return f.Type.OfType.OfType.OfType.OfType.OfType.Name
+	} else if f.Type.OfType.OfType.OfType.OfType.OfType.OfType.Name != "" {
+		return f.Type.OfType.OfType.OfType.OfType.OfType.OfType.Name
+	}
+
+	return ""
+}
+
+// fieldTypeFromTypeRef resolves the given SchemaInputValue into a field name to use on a go struct.
+func fieldTypeFromTypeRef(f SchemaInputValue) (string, bool, error) {
+
+	switch t := firstTypeName(f); t {
+	case "String":
+		return "string", false, nil
+	case "Int":
+		return "int", false, nil
+	case "Boolean":
+		return "bool", false, nil
+	case "Float":
+		return "float64", false, nil
+	case "":
+		return "", true, fmt.Errorf("empty field name: %+v", f)
+	}
+
+	return "", true, fmt.Errorf("need to handle Field f: %+v", f)
+}
+
+func handleInputType(schema Schema, t SchemaType) map[string]string {
+	types := make(map[string]string)
+	var err error
+	recurse := false
+
+	// output collects each line of a struct type
+	output := []string{}
+
+	output = append(output, fmt.Sprintf("type %s struct {", t.Name))
+
+	// Fill in the struct fields for an input type
+	for _, f := range t.InputFields {
+		var fieldType string
+
+		fmt.Printf("handling kind %s: %+v\n\n", f.Type.Kind, f)
```

Comment:
`log.Printf` for stderr?

Reply:
Oh yeah, I wasn't sure the right way to get things to print due to being run as an integration test, though I think we probably want to move it out of a test and into a make target or something.  Just not sure where to locate the code that will do the command of generating the files.